integration/docker/env_test.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	. "github.com/clearcontainers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("env", func() {
+	var (
+		args     []string
+		id       string
+		exitCode int
+		command  *Command
+	)
+
+	runCommand := func(args []string, expectedExitCode int) {
+		command = NewCommand(Docker, args...)
+		Expect(command).ToNot(BeNil())
+		exitCode = command.Run()
+		Expect(exitCode).To(Equal(expectedExitCode))
+	}
+
+	AfterEach(func() {
+		Expect(ContainerRemove(id)).To(BeTrue())
+		Expect(ContainerExists(id)).NotTo(BeTrue())
+	})
+
+	Describe("check env", func() {
+		Context("check that required env variables are set", func() {
+			It("should have path, hostname, home", func() {
+				id = RandID(30)
+				args = []string{"run", "--name", id, Image, "env"}
+				runCommand(args, 0)
+				Expect(command.Stdout.String()).To(ContainSubstring("PATH"))
+				Expect(command.Stdout.String()).To(ContainSubstring("HOME"))
+				Expect(command.Stdout.String()).To(ContainSubstring("HOSTNAME"))
```

Comment:
I guess docker sets the ```id``` as hostname isn't it? can you check that ```HOSTNAME``` is equal to ```id```?